src/script/descriptor.cpp

```diff
@@ -298,6 +298,14 @@ class ConstPubkeyProvider final : public PubkeyProvider
     CPubKey m_pubkey;
     bool m_xonly;
 
+    std::optional<CKey> GetPrivKey(const SigningProvider& arg) const
```

Comment:
Hm, I am not seeing why having two `GetPrivKey`s is better than one. The description of the commit only talks about the other one, not this one. For this one the caller still cares about the return value it seems. Can't they be unified under the `std::optional` returning variant? It seems like the variant using `FlatSigningProvider` is just used in one place in `ExpandPrivate` and it just saves us one line there.

Reply:
The benefit is in the MuSig2 PRs where `GetPrivKey` can insert multiple private keys to the `FlatSigningProvider`, none of which are the private key for the MuSig2 aggregate pubkey itself.

For the `ConstPubkeyProvider`, it's useful to have another `GetPrivKey` that returns the single private key since this is behavior is used in a few places within this class's member functions.

Reply:
Understood. I think I would have chosen a different name for the `ConstPubkeyProvider`, like `GetConstPrivKey` maybe, but completely optional.

Reply:
> The benefit is in the MuSig2 PRs where GetPrivKey can insert multiple private keys to the FlatSigningProvider, none of which are the private key for the MuSig2 aggregate pubkey itself.

Something around this can be mentioned either in the commit message and/or the PR description.

Reply:
>  like GetConstPrivKey maybe

This is already inside `ConstPubkeyProvider` class, using `Const` again could feel repetitive. I think the function overloading (with different argument types) is differentiating enough. Though, some documentation for the new function would be helpful for the reader.

Reply:
> Something around this can be mentioned either in the commit message and/or the PR description.

Added to the commit message.